src/Microsoft.ML.PipelineInference/AutoInference.cs

```diff
@@ -524,6 +431,18 @@ public void AddEvaluated(PipelinePattern pipeline)
                     d += 1e-3;
                 _sortedSampledElements.Add(d, pipeline);
                 _history.Add(pipeline);
+
+                using (var ch = _host.Start("Suggested Pipeline"))
+                {
+                    ch.Info($"PipelineSweeper Pipeline Id : {pipeline.UniqueId}");
+                    foreach (var transform in pipeline.Transforms)
+                    {
+                        ch.Info($"PipelineSweeper Transform : {transform.Transform}");
+                    }
+                    ch.Info($"PipelineSweeper Learner : {pipeline.Learner}");
+                    ch.Info($"PipelineSweeper Train Metric Value: {pipeline.PerformanceSummary.TrainingMetricValue}");
+                    ch.Info($"PipelineSweeper Test Metric Value: {pipeline.PerformanceSummary.MetricValue}");
```

Comment:
We're printing the primary sweeping metric here. How much work would it be to print all of the pipeline's metrics?

Generally when hand sweeping over a dataset, I'm watching multiple metrics. This is important as one metric often increases at the detriment of others. 

For instance, micro & macro accuracy often trade-off as some models end-up better at smaller classes or the majority classes. The best models improve the metrics in unison.

Reply:
Good idea. I need to explore how we can expose this functionality  for the PipelineSweeper.  One thing to note is that the other metrics would depend on the TrainerKind. 

I will track this ask as a separate task. Would not want to block this PR on this feature ask. 

---
In reply to: [203264334](https://github.com/dotnet/machinelearning/pull/539#discussion_r203264334) [](ancestors = 203264334)